File: core/src/projection/core/Feed.cpp
#include "projection/core/Feed.h"

namespace projection::core {

Feed::Feed(ProjectId projectId, FeedId id, std::string name, AssetId assetId, FeedSettings settings)
    : projectId_(std::move(projectId)),
      id_(std::move(id)),
      name_(std::move(name)),
      assetId_(std::move(assetId)),
      settings_(std::move(settings)) {}

Feed makeFeed(const ProjectId& projectId, const FeedId& id, const std::string& name, const AssetId& assetId,
              const FeedSettings& settings) {
  return Feed(projectId, id, name, assetId, settings);
}

}  // namespace projection::core

File: clients/composer/src/types/feed.ts
export type PanDirection = "leftToRight" | "rightToLeft" | "pingPong";

export type FeedSettings = {
  variantPath?: string;
  monochrome?: boolean;
  panDirection?: PanDirection;
  panDurationSeconds?: number;
  visiblePortion?: number;
};

export const defaultFeedSettings: Required<FeedSettings> = {
  variantPath: "",
  monochrome: false,
  panDirection: "leftToRight",
  panDurationSeconds: 120,
  visiblePortion: 0.6,
};

export const panDirectionLabels: Record<PanDirection, string> = {
  leftToRight: "Left to Right",
  rightToLeft: "Right to Left",
  pingPong: "Ping Pong (Back & Forth)",
};

export type Feed = {
  projectId: string;
  id: string;
  name: string;
  assetId: string;
  settings?: FeedSettings;
};

